opendevin/controller/agent_controller.py

```diff
@@ -110,6 +110,11 @@ async def _run(self):
                 await self.notify_task_state_changed()
                 break
 
+            if self._is_stuck():
+                logger.info('Loop detected, stopping task')
```

Comment:
maybe we should stick an `AgentErrorObservation` or similar in here, so the user gets some feedback that a loop was detected